File: benchmarks/NSubstitute.Analyzers.Benchmarks.Source.VisualBasic/Models/IFoo.vb
Imports NSubstitute.Core

Namespace Models
    Public Interface IFoo
        Sub VoidReturningMethod()
        
        Sub VoidReturningMethodWithArguments(ByVal a As Integer, ByVal b As Integer, ByVal c As Decimal)

        Function ObjectReturningMethod() As IFoo
        
        Function ObjectReturningMethodWithArguments(ByVal a As Integer, ByVal b As Integer, ByVal c As Decimal) As IFoo
        
        Function ObjectReturningMethodWithRefArguments(ByRef a As Integer, ByRef b As Integer, ByRef c As Decimal) As IFoo
        
        Function CallInfoReturningMethod() As ConfiguredCall
        
        ReadOnly Property ConfiguredCallReturningProperty As ConfiguredCall
        
        ReadOnly Property IntReturningProperty As Integer
        
        ReadOnly Property ObjectReturningProperty As IFoo
        
        Default ReadOnly Property Item(ByVal a As Integer) As Integer
        
        Default ReadOnly Property Item(ByVal a As Integer, ByVal b As Integer) As IFoo
        
        Default ReadOnly Property Item(ByVal a As Integer, ByVal b As Integer, ByVal c As Integer, ByVal d As Integer) As ConfiguredCall
        
    End Interface
End Namespace

File: benchmarks/NSubstitute.Analyzers.Benchmarks.Source.CSharp/Models/IFoo.cs
using NSubstitute.Core;

namespace NSubstitute.Analyzers.Benchmarks.Source.CSharp.Models
{
    public interface IFoo
    {
        void VoidReturningMethod();

        void VoidReturningMethodWithArguments(int a, int b, decimal c);

        IFoo ObjectReturningMethod();

        IFoo ObjectReturningMethodWithArguments(int a, int b, decimal c);

        IFoo ObjectReturningMethodWithRefArguments(ref int a, ref int b, ref decimal c);

        ConfiguredCall CallInfoReturningMethod();

        ConfiguredCall ConfiguredCallReturningProperty { get; }

        int IntReturningProperty { get; set; }

        IFoo ObjectReturningProperty { get; }

        int this[int a] { get; }

        IFoo this[int a, int b] { get; }

        ConfiguredCall this[int a, int b, int c, int d] { get; }
    }
}